File: benchmark/c/opl/calc_vol/calc_vol.c
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>
#include <string.h>

static void calc_vol(unsigned char *regbyte, int volume) ;
static void calc_vol(unsigned char *regbyte, int volume) {
  int level;
  level = ~(*regbyte);
  level &= 0x3f;
  level = (level * volume) / 127;
  if (level > 0x3f) level = 0x3f;
  if (level < 0) level = 0;
  level = ~level;
  level &= 0x3f;
  *regbyte &= 0xC0;
  *regbyte |= level;
}

File: translation_result/real_world_code/fail/c/opl/calc_vol/gemini/Hinted/calc_vol.rs

fn calc_vol(regbyte: &mut u8, volume: i32) {
    let mut level = !(*regbyte);
    level &= 0x3F;
    level = ((level as i32 * volume) / 127) as u8;
    if level > 0x3F {
        level = 0x3F;
    }
    if level < 0 {
        level = 0;
    }
    level = !level;
    level &= 0x3F;
    *regbyte &= 0xC0;
    *regbyte |= level;
}
